app/src/main/java/com/portfolio/tasky/entry/network/EntryApiCall.kt

```diff
@@ -0,0 +1,17 @@
+package com.portfolio.tasky.entry.network
+
+import com.portfolio.tasky.entry.models.AuthenticationRequest
+import com.portfolio.tasky.entry.models.AuthenticationResponse
+import com.portfolio.tasky.entry.models.RegisterRequest
+import retrofit2.Response
+import retrofit2.http.Body
+import retrofit2.http.POST
+
+interface EntryApiCall {
+
+    @POST("/register")
+    suspend fun registration(@Body registerModel: RegisterRequest) : Response<Void>
+
+    @POST("/login")
+    suspend fun login(@Body authenticationModel: AuthenticationRequest) : Response<AuthenticationResponse>
```

Comment:
No need to wrap it in a "Response" class

You could simply return the object

Ex: On login: AuthenticationResponse
And on Registration:  ResponseBody (This is used instead of Void, so you can check if it returned a 200, 404, etc)

Reply:
At this stage this is making my implementation look simple. I can use the result as Response<> and process it in one place whether success or failure and return the data in case of success. Very easy as per my understanding. Will replace it with better version  in future.